contracts/ERC20AssetHolder.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+import './Outcome.sol';
+import './AssetHolder.sol';
+
+contract IERC20 {
+    // Abstraction of the parts of the ERC20 Interface that we need
+    function transfer(address to, uint256 tokens) public returns (bool success);
+    function transferFrom(address from, address to, uint256 tokens) public returns (bool success);
+}
+
+contract ERC20AssetHolder is AssetHolder {
+    address AdjudicatorAddress;
+    address TokenAddress;
+
+    constructor(address _AdjudicatorAddress, address _TokenAddress) public {
+        AdjudicatorAddress = _AdjudicatorAddress;
+        TokenAddress = _TokenAddress;
+    }
+
+    modifier AdjudicatorOnly {
```

Comment:
I might be missing something, but it looks like this modifier is not used.

Reply:
It does appear that way; however, the modifier is used in the base contract `AssetHolder` that this contract extends. It is also defined there, but needs to be re-defined here in order to  create the constructor correctly with the `AdjudicatorAddress` (only known at deploy-time). 

Take a look at [this branch](https://github.com/statechannels/nitro-protocol/compare/asset-holder...adjudicator-only-modifier?expand=1) where I removed the modifier -- tests fail because the modifier is being defined only in the base contract, and hence before the `AdjudicatorAddress` is defined.